_pages/develop/map.md

```diff
@@ -0,0 +1,682 @@
+zoomed out, and to gradually turn on more collections as the user zooms in.  This
+requires the storage of the data to be considered as part of the map development
+process, as display needs may impact the design of the storage solution.
+
+## Example: displaying additional map data as the user zooms in
+The following image shows a zoomed-out map displaying just substation
+locations and high voltage conductors in an electrical network:
+
+<img src="../../img/developer-guides/map/map-zooming-1.png"/>
+
+When the user zooms in, transformers and medium voltage conductors are made visible:
+
+<img src="../../img/developer-guides/map/map-zooming-2.png"/>
+
+When the user zooms in further, switches, circuit breakers, and low voltage
+conductors are made visible as well:
+
+<img src="../../img/developer-guides/map/map-zooming-3.png"/>
+
+## Rendering objects directly on the HTML canvas
+If, after all these steps have been taken, the performance of the map still does
+not meet your needs, then you may want to consider bypassing the Leaflet rendering
+code that uses DOM elements to display and track the location of objects, and
+instead, render the objects directly on an HTML canvas. There is a lot of development
+effort involved in doing this, as you not only have to reimplement the rendering
+behavior but also the ability to determine what, if any, objects the user has
+clicked on in the map. However, this approach will give you the best performance
+with large volumes of data.
+
+# Rendering data in a specific order
+By default, Leaflet.js uses three DOM elements -- points, lines, and areas --
+for rendering the bulk of the objects on the map based on their geometry. Points
+are rendered on top of lines, and lines are rendered on top of areas. This basic
+approach works most of the time, but there are instances when the map requires a specific rendering order.
+
+In the following map, it was important to ensure that the transformers (blue
+  triangles) were rendered on top of all the other point objects:
+
+<img src="../../img/developer-guides/map/map-data-rendering.png"/>
+
+Leaflet.js would have rendered all the points as part of the same DOM element,
+in the order that they were added to that element. As the data is returned
+asynchronously, the rendering order is likely to be different every time the map
+is drawn.
+
+When more control of the rendering order is required, it is possible to add the
+data as Leaflet.js custom panes. Custom panes can be created with their own z-index,
+which can be used to force a specific rendering order. For complicated data like
+an electrical network, it may be necessary to have specific custom panes for each
+object type.
+
+# Other features
+
+## Dropping a marker at the user's current location
+
+Use the `px-map-marker-locate` component together with the `px-map-control-locate`
+component to automatically find and set the latitude and longitude of the marker
+to the user's current location:
+
+    ```html
```

Comment:
Same as above, formatting off.

Reply:
Can you attach a screenshot? Works on my machine 😉

Reply:
All the ones that are indented show up like this for me:

<img width="455" alt="screen shot 2018-03-01 at 2 57 15 pm" src="https://user-images.githubusercontent.com/2130613/36874470-d8c799cc-1d60-11e8-9671-2adcb0f3f931.png">